File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    for i in range(1, int(yellow ** 0.5) + 1):
        w, h = yellow // i, i
        if yellow % h == 0 and 2 * (w + h) == brown - 4:
            return [w + 2, h + 2]

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {0, 0};
        for (int i = 1; i < (int) Math.sqrt(yellow) + 1; i++) {
            int w = (int) yellow / i; int h = i;
            if (yellow % h == 0 && 2 * (w + h) == brown - 4) {
                answer[0] = w + 2; answer[1] = h + 2;
                return answer;
            }
        }
        return answer;
    }
}